libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
We saw that despite that openat was called the kernel driver could destroy and re-create its uverbs file (e.g. uverbs0) and as such we didn't find any added value by using this system call.

We'll go with the TS check solution and upload a version once be ready.
Re errno: EIO will be returned in case TS was changed after opening the cmd_fd or in case the device was dropped as part of the alloc context command that follows the open.

As such, application can know that this was the case of device removal when this errno was set.